cockpit/src/subscriptions-client.js

```diff
@@ -274,6 +279,30 @@ client.getSubscriptionStatus = () => {
     });
 };
 
+client.readConfig = () => {
+    return configService.wait(() => configService.GetAll().then(config => {
+        const usingDefaultUrl = (config.server.v.port === '443' &&
+            config.server.v.hostname === 'subscription.rhsm.redhat.com' &&
+            config.server.v.port === '443' &&
+            config.server.v.prefix === '/subscription'
+        );
+        const maybePort = config.server.v.port === '443' ? '' : config.server.v.port;
+        const maybePrefix = config.server.v.prefix === '/subscription' ? '' : config.server.v.prefix;
+        const serverUrl = usingDefaultUrl ? '' : `${config.server.v.hostname}${maybePort}${maybePrefix}`;
```

Comment:
There is missing colon separator between hostname and port:
```javascript
const serverUrl = usingDefaultUrl ? '' : `${config.server.v.hostname}:${maybePort}${maybePrefix}`;
```
In theory IPv6 address can be used. In this case IPv6 address should be encapsulated in `[ ]` brackets.

Reply:
Insightful! Will address both.